glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -2147,14 +2116,21 @@ where
                 offset,
                 count,
                 route,
+                sub_pipeline,
             } => {
-                Self::try_pipeline_request(
-                    pipeline,
-                    offset,
-                    count,
-                    Self::get_connection(route, core, None),
-                )
-                .await
+                if pipeline.is_atomic() || sub_pipeline {
+                    // If the pipeline is atomic (i.e., a transaction) or if the pipeline is already splitted into sub-pipelines, we can send it as is, with no need to split it into sub-pipelines.
+                    Self::try_pipeline_request(
+                        pipeline,
+                        offset,
+                        count,
+                        Self::get_connection(route, core, None),
+                    )
+                    .await
+                } else {
+                    // The pipeline is not atomic and not already splitted, we need to split it into sub-pipelines and send them separately.
+                    Self::handle_pipeline_request(&pipeline, core).await
```

Comment:
handle_pipeline_request -> handle_multi_slot_pipeline_request / handle_non_atomic_pipeline or something like that.
now the code is confusing -
```
if atomic {
try_pipeline_request ()
} else {
handle_pipeline_request()
}